orca-core/src/main/groovy/com/netflix/spinnaker/orca/batch/adapters/TaskTasklet.groovy

```diff
@@ -173,9 +177,17 @@ class TaskTasklet implements Tasklet {
     }
 
     try {
-      def currentUser = new User(
+      // An AuthenticatedStage can override the default pipeline authentication credentials
+      def authenticatedUser = stageNavigator.findAll(stage, { Stage ancestorStage, StageBuilder stageBuilder ->
```

Comment:
its not clear what the ordering would be here (its it finding the stages by traversing backwards up the graph to the nearest ancestor?) If that is the case maybe adding a findNearestAncestor on stageNavigator would be useful